js/main.js

```diff
@@ -1,255 +1,4 @@
-            }
-            if (_.random(1, 100) <= 25) {
-                label += ' of the ' + _.sample(vocabulary.internet) + '.';
-            } else {
-                label += '.';
-            }
-    }
-    return label;
-};
-
-//
-// Good Morning Scum
-//
-// Produces a quote like: "Good Morning, Bastards of the Internet"
-//
-var goodMorningScum = function(vocabulary) {
-    var greeting = '';
-
-    // eg, Good Morning
-    greeting += _.sample(vocabulary.salutation) + ' ';
-
-    // eg, "bastards of the internet"
-    greeting += ellisLabel(vocabulary);
-
-    if (_.random(1, 100) <= 10) {
-        // eg, "I hate you"
-        greeting += ' ' + _.sample(vocabulary.extra);
-    }
-
-    return greeting;
-};
-
-//
-// Attention Scum
-//
-// Produces a quote like "ATTENTION SCUM: I will destroy you"
-//
-var attentionScum = function(vocabulary) {
-    var greeting = '';
-
-    if (_.random(1, 100) <= 50) {
-        // eg, "ATTENTION SCUM"
-        greeting += _.sample(vocabulary.attention) + ' ';
-
-    } else {
-        // eg, "ATTENTION $FILTH" with a random filth name
-        greeting += "attention ";
-        greeting += _.sample(vocabulary.attentive_filth) + ': ';
-    }
-
-    // uppercase the "ATTENTION SCUM" part
-    greeting = greeting.toUpperCase();
-
-    // eg, "I hate you"
-    greeting += _.sample(vocabulary.extra);
-
-    return greeting;
-};
-
-//
-// Build a Quote
-//
-// This is our final function. It chooses between two paths:
-// 1. 85% chance to call goodMorningScum(), and an
-//    addtional 10% chance to append an extra bit at the end.
-// 2. 15% chance to call attentionScum()
-//
-var buildQuote = function(vocabulary) {
-    var quote = '';
-
-    if (_.random(1, 100) <= 85) {
-        // eg, "Good Morning, Scum"
-        quote = goodMorningScum(vocabulary);
-
-    } else {
-        // eg, "ATTENTION SCUM: I hate you"
-        quote = attentionScum(vocabulary);
-    }
-
-    return quote;
-};
-
-// Prepare the Ajax request
-xhr.open('GET', 'data/vocabulary.json', true);
-
-// Handle the Ajax response
-xhr.onload = function() {
-
-    // We got a good response from the server
-    if (xhr.status >= 200 && xhr.status < 400) {
-
-        // Parse the JSON data
-        var vocabulary = JSON.parse(xhr.responseText);
-        document.getElementById('quote').innerHTML = buildQuote(vocabulary);
-
-    // We reached our target server, but it returned an error
-    } else {
-        var error = 'Whoops! Something went wrong. Please try again.';
-        document.getElementById('datelist-error').innerHTML = error;
-    }
-};
-
-// Handle an Ajax connection error
-xhr.onerror = function() {
-    var error = 'Whoops! We couldn\'t reach the server. Please try again.';
-    document.getElementById('datelist-error').innerHTML = error;
-};
-
-// Send the Ajax request
-xhr.send();
+var vocabulary = loadData('data/vocabulary.json');
+document.getElementById('quote').innerHTML = talkLikeWarrenEllis(vocabulary);
```

Comment:
Ahhh... Thank you, that makes sense. I was pretty tired last night and I was spinning up some sort of conspiracy theory like I couldn't pass data from one function to the other and thought I was going crazy. Good to know it's a more simple problem.

You know what's funny? I went 90% of the way down implimenting a Promise last night after I found a really good example, but I backed off when I realized there's no IE support, even in IE11.

Does the callback method you mention above have any browser support problems? It looks like it should work everywhere.

Alternately, I could go all-in on writing a promise, given that this is a person project that doesn't need IE support.

What do you think?